### DISCUSSION TITLE
refactor: It is aimed to remove the feature of renewing the data in the page at certain time intervals.

### DISCUSSION THREAD

**Discussion by khnbzkrt** [acikkaynak/deprem-yardim-frontend#632] [Ideas]
### Refactor Request

** discord kullanıcı adı: Hakan Angın#9836 **

### Is your refactor request related to an issue? Please explain.

We can update the page's data with the "Scan Field" button. At the same time, the data in the counter is updated automatically at certain time intervals. However, this automatic update process is both a non-functional development feature and it is perceived as a bug that the initial value is 30 seconds and the counter start is 900 seconds. The high matter of 900 seconds also causes confusion and becomes a different bug issue. Since this standard cannot be established and considering its benefit, it is planned to remove this feature.

### Additional Context
![image](https://user-images.githubusercontent.com/67371416/217949658-6232e49b-d1a4-40f9-b24d-6559dca5dded.png)

**Comment by eraygundogmus** [comment]
@usirin karar verebilir misin? daha sonrasında labelları güncellemeni rica ediyorum

**Comment by keislamoglu** [comment]
Şu anda artış formülü `x *= x` şeklinde. İlk bekleme süresi **30** saniye olduğu için **30**, **30*30**, **900*900**, şeklinde gidiyor. Birden bire süre oldukça artıyor ve neredeyse ikinci requestten sonra bekleme süresi anlamsızlaşıyor. Burada artışı daha iyi formülize ederek optimum bir çözüme gidilebilir. Mesela artışın `x *= 3` şeklinde düzenlenmesiyle **30**, **30*3**, **90*3**, **270*3** gibi daha anlamlı bir sonuç alınabilir.

**Comment by keislamoglu** [comment]
Bu özellik eklenirken hedeflenen, ekranı açık bir şekilde aynı bölgeyi inceleyen kişiler/kurumlar olabilir, dolayısıyla sürekli butona basmak durumunda olmasınlar idi.

**Comment by aerdemsenturk** [comment]
Aslında alanı tara özelliği başlarda boundingbox dışındaki noktaları hariç tutarak json boyutunu küçük tutmak içindi diye hatırlıyorum. Şimdi ise json boyutu zaten çok küçülmüş, noktaların tamamı da çağrılsa çok hızlı geliyor. 

Dolayısıyla en başta tüm noktalar tek seferde gelse, sürekli istek yapmaya da bu "Alanı tara" butonuna da hiç gerek kalmayacak.

<img width="1210" alt="Ekran Resmi 2023-02-10 01 09 23" src="https://user-images.githubusercontent.com/25268260/217967503-08a5a29a-9bed-4316-8e47-31a5e6bb56f3.png">

**Comment by keislamoglu** [comment]
@aerdemsenturk yine de sayfayı canlı tutmak için aralıklı request atılmasında fayda görüyorum. Aksi halde sayfayı refreshlemek gerekir güncel data için.

**Comment by aerdemsenturk** [comment]
Doğru. Ama yine de butonlar biraz boşa yer kaplıyor hale geliyor orada. Refresh işini sol tarftaki buton yapabilirdi. Sayacı ise saniye saniye bilmemize gerek yok diye düşünüyorum. 

3dk'da bir aralıklı request atılsa ama boundingbox yerine her seferinde tümünü çağırsa, böylece kullanıcı haritada sağa sola gezindikçe tekrar alanı tara butonuna basmasına gerek kalmaz. 

"Son güncelleme x dk önce" gibi kenarda yazar saymak yerine. Üst taraf biraz daha ferahlamış olurdu. Daha minimal bir çözüm ve aynı kapıya çıkıyor.

**Comment by yuskraft** [comment]
Bu konuda bir karar verildi mi?

**Comment by cdagli** [comment]
Why should we remove the auto refresh feature? 

1) Refresh timer label takes too much space on small screens: 
<img width="374" alt="image" src="https://user-images.githubusercontent.com/7020369/218255909-78ef7641-f47e-4908-b6f6-e420a2937c89.png">

2) After user fetches a larger area and zooms in, auto-refresh triggers another fetch and makes the previously fetched data unusable. 

3) It increases the code complexity and increases the effort to maintain. 
https://github.com/acikkaynak/deprem-yardim-frontend/issues/481
https://github.com/acikkaynak/deprem-yardim-frontend/issues/695
https://github.com/acikkaynak/deprem-yardim-frontend/issues/607

4) It does not add any user value. It was meaningful in the beginning when we had performance issues but now we are not having those. 

We can still throttle the button and make sure user does not click the button on the same location multiple times in N seconds time period.
